lib/ansible/__main__.py

```diff
@@ -0,0 +1,21 @@
+# Copyright: (c) 2021, Matt Martz <matt@sivel.net>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+import argparse
+from importlib.metadata import distribution
+
+
+def main():
+    dist = distribution('ansible-core')
+    ep_map = {ep.name: ep for ep in dist.entry_points if ep.group == 'console_scripts'}
+
+    parser = argparse.ArgumentParser(prog='python -m ansible', add_help=False)
```

Comment:
It's not going to always be `python`. Sometimes it's `python3.10` or `python3` or `./path/to/venv/bin/python`.
Maybe it could be something along the lines of
```suggestion
    mod_path = Path(__file__)
    if mod_path.parts[-1] == '__main__.py':
        mod_path = f'{sys.executable} -m {(mod_path / "..").resolve().parts[-1]}'
    parser = argparse.ArgumentParser(prog=str(mod_path), add_help=False)
```

Reply:
meh, it's close enough for me.  I'd rather not do gymnastics to figure it out.

Reply:
I'm not going to push for this. Just remembered one case where `pip` (with a Fedora patch on top) used `__main__.py` in their output and _that_ was rather confusing.